homeassistant/components/bluetooth/manager.py

```diff
@@ -110,12 +118,18 @@ def _discover_service_info(self, service_info: BluetoothServiceInfoBleak) -> Non
             except Exception:
                 _LOGGER.exception("Error in bluetooth callback")
 
+        discovery_key = discovery_flow.DiscoveryKey(
```

Comment:
```suggestion
        if not matched_domains:
            return # avoid creating DiscoveryKey if there are no matches

        discovery_key = discovery_flow.DiscoveryKey(
```

Reply:
OK, makes sense. I think we should do the same also in the other discovery integrations, although I expect they are less busy?

Reply:
It probably only matters for bluetooth/zeroconf/ssdp (maybe dhcp .. but only in rare cases) as everything else should be lower volume but doesn't hurt to do it everywhere